projects/distributed-tracing/src/shared/dashboard/widgets/waterfall/waterfall/waterfall-chart.service.ts

```diff
@@ -123,7 +123,10 @@ export class WaterfallChartService {
       const node = nodes.shift()!;
       let color;
 
-      if (colorMap.has(node.serviceName)) {
+      if (node.$$spanName.error) {
```

Comment:
nit: the color logic can be isolated to its own method now (since it has gotten bigger).

Reply:
I guess, we don't need this as of now.
It is not that big and we also need to handle some local vars(the way code is written currently)